tools/deploy-prod.sh

```
@@ -47,14 +55,113 @@ fi
+sys.exit(0)
+' "$@" || exit 1
 
 # -----------------------------------------------------------------------------
 # 3. SUDO LOCK
 # -----------------------------------------------------------------------------
 echo -e "\n${YELLOW}[3/3] Enforcing Sudo Lock...${NC}"
 echo -e "${YELLOW}[+] Verifying playbook and roles configurations...${NC}"
 
+# Verify that play-level become is not set to true in deploy_prod_compose.yml
+# under set -e we use grep | wc -l to safely capture counts as an integer expression
+become_count=$(grep -E "^[[:space:]]{2}become:[[:space:]]*true" playbooks/deploy_prod_compose.yml | wc -l)
```

Shellcheck (0.11.0)

[style] Consider using 'grep -c' instead of 'grep|wc -l'.

(SC2126)